specs/deposits.md

```diff
@@ -0,0 +1,272 @@
+# Specification for deposits functionality.
+
+### Motivation
+
+We want solution for the following problem:
+
+1. User A wants to use the Golem Network to perform a computation, but doesn't want to run yagna.
+2. User A wants person B to run yagna on their behalf.
+3. User A wants to give allowance for spending his funds to person B.
+4. Person B wants to be sure that User A will pay for the computation.
+5. Person B wants to be paid for being middle man (for example cover gas costs and/or earn some money)
+
+Funder (User A) - Person that is funding the deposit and requesting job from the Spender service
+
+![funder](deposit_assets/funder300.webp)
+
+Spender (Person B) - Service based on yagna
+
+![spender](deposit_assets/spender300.webp)
+
+
+### Solution attempt allowance
+
+The simple solution is to use ERC20 allowance to person B's funds, but this solution has some drawbacks:
+Point 4 is not satisfied, because User A can revoke allowance at any time.
+Point 5 can be satisfied by taking funds beforehand.
+
+### Solution 
+
+Contract that will hold funds until specified time.
```

Comment:
It would be good to have generalized description here, before we show contract details. And I would move flow chart somewhere here. Documentation should rather take top-down approach